packages/core/src/typegenTypes.ts

```diff
@@ -146,32 +149,16 @@ export interface MarkAllImplementationsAsProvided<TResolvedTypesMeta> {
   resolved: Prop<TResolvedTypesMeta, 'resolved'> & AllImplementationsProvided;
 }
 
-type GenerateActorEvent<
-  TActorName,
-  TEventType,
-  TActorMap extends ActorMap
-> = TEventType extends any
+type GenerateActorEvents<
+  TActor extends ProvidedActor,
+  _TInvokeSrcNameMap
+> = TActor extends { id: infer K }
   ? {
-      type: TEventType;
-    } & Prop<TActorMap, TActorName>
+      type: `done.invoke.${K & string}`;
+      output: OutputFrom<TActor['logic']>;
+    }
   : never;
 
-type GenerateActorEvents<
-  TActorMap extends ActorMap,
-  TInvokeSrcNameMap
-> = string extends keyof TActorMap
-  ? never
```

Comment:
TODO: this had a potentially important check here, need to re-evaluate its role and if we are losing anything by removing this today

Reply:
This was still needed for some things that we do. I brought this in https://github.com/statelyai/xstate/pull/4036/commits/d5b63e33a4a3a903c572ac20458a1884af2b734a and added comments explaining this whole type in https://github.com/statelyai/xstate/pull/4036/commits/ca81cc66e2a4d3bf3a6ba4f44fa19a4f0dbdd07f